src/main/java/seedu/address/model/person/Person.java

```diff
@@ -75,15 +99,22 @@ public Birthday getBirthday() {
         return Collections.unmodifiableSet(tags);
     }
 
-
     public List<Event> getDates() {
         return Collections.unmodifiableList(dates);
     }
 
+    public Person setDates(List<Event> dates) {
```

Comment:
If we were to set Person as immutable, setDates, setMeetings and setPicture might need to shift to a different class instead. Maybe a PersonFactory?

Either this or state in comments there are mutable fields present in Person.

Reply:
Person is still immutable, the setters are not changing the Person itself. Will change it to `withDates` instead of `setDates` because the `set` prefix is commonly associated with mutability.